### PR TITLE
fix(a11y): add aria-describedby for clrInput

### PR DESCRIPTION
**Pull request by bdryanovski**
Add `aria-describedby` to `clrInput` in the context of `clr-input-container` to provide additional description when there is a error text.

## PR Checklist

Please check if your PR fulfills the following requirements:

* [x] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

## Other information

Close: #3562

### CODE REVIEW

File: src/clr-angular/forms/input/input.spec.ts [hunk 1]
```diff
@@ -24,6 +24,17 @@ class StandaloneUseTest {}
...
 
+@Component({
+  template: `
+    <clr-input-container>
+       <input clrInput name="model" [aria-describedby]="describedby" class="test-class" [(ngModel)]="model" />
+    <clr-input-container>
+    `,
+})
+class InputContainerTest {
+  describedby: string;
```
Comment (gnomeontherun) [inline src/clr-angular/forms/input/input.spec.ts]: What are you testing here with an empty string?

File: src/clr-angular/forms/input/input.ts [hunk 2]
```diff
@@ -21,8 +28,24 @@ export class ClrInput extends WrappedFormControl<ClrInputContainer> {
...
+    el: ElementRef,
+    @Optional() public controlIdService: ControlIdService
   ) {
     super(vcr, ClrInputContainer, injector, control, renderer, el);
   }
+
+  /**
+   * Provide a way to overwrite aria-describedby property.
+   */
+  @Input('aria-describedby') userDescribedById: string;
```
Comment (gnomeontherun) [inline src/clr-angular/forms/input/input.ts]: This is going to cause problems with the logic in wrapped-control.

**Comment by Shijir** [comment]
@bdryanovski haven't we solved #3562 with the AriaLive service? If the solution provided with the AriaLive service wasn't sufficient, what particular issue is this one is solving? Have we checked with @chlane on this solution?

**Comment by bdryanovski** [comment]
It seems that my PR is no longer valid - the `aria-describedby` is applied as expected. The only issue that I find here is that AriaLiveService is not working with `clr-input-error` when is inside the `clr-input-wrapper` but this is separate bug #4222  

So I'm closing this one.

**Comment by clane** [comment]
@Shijir I recommended using both a live region and aria-describedby together for error handling https://github.com/vmware/clarity/issues/3562. I recommend using aria-invalid as well. See my recommendations for error handling below.

After considering the sequential nature of screen reader usage and the potential for a high volume of error messages in our applications, we recommend the following.

Use a polite live region or role="status" to announce a status message that lists the fields in error e.g. "There are errors with the following fields: Memory, New Hard Disk, NEW SCSI controllere" for more information see https://www.w3.org/TR/wai-aria-1.1/#aria-live. Do not move focus to this element.
Use aria-label="error:" and role="img" with the error icons to make these icons be announced as "error:" to blind screen reader users.
Provide visible textual error messages following the error icon e.g. "memory allocation is too low".
Use aria-describedby (https://www.w3.org/TR/wai-aria-1.1/#aria-describedby) to programmatically associate the error icons and error messages with their respective fields. This will allow blind screen reader users to hear the name of a given input field, the word "error" and the respective error message together whenever they read a given input field.
Use aria-invalid (https://www.w3.org/TR/wai-aria-1.1/#aria-invalid) when the value entered by the user invalid or out-of-range. When the user attempts to submit data involving a field for which aria-required is true, authors MAY use the aria-invalid attribute to signal there is an error. However, if the user has not attempted to submit the form, authors SHOULD NOT set the aria-invalid attribute on required widgets simply because the user has not yet entered data.
If there is a single error, focus should be moved to that field
If there are multiple errors, focus should be moved to first field in error
If any controls need to be disabled, do so both visually and programmatically i.e,. use the disabled attributed
Provide a checkbox to allow blind screen reader users to disable the status messages if the application becomes too verbose
Provide configuration options for the status messages allowing blind screen reader users the ability to control the verbosity of the status message in cases where it may be too verbose

**Pull request closed without merging**